csharp/ql/lib/semmle/code/csharp/dataflow/internal/FlowSummaryImpl.qll

```diff
@@ -137,9 +143,13 @@ module Public {
     or
     noComponentSpecificCsv(sc) and
     (
-      exists(int i | sc = TParameterSummaryComponent(i) and result = "Parameter[" + i + "]")
+      exists(ArgumentPosition pos |
+        sc = TParameterSummaryComponent(pos) and result = "Parameter[" + pos + "]"
```

Comment:
This mixes up the `toString` of `ArgumentPosition` with its csv representation, which seems slightly wrong. Perhaps `ArgumentPosition` ought to have a `toSpec` with suitable qldoc (and then the `toString` could alias that if need be) to clarify that its about more than the usual prettyprinting.
Same goes for `ParameterPosition`.
```suggestion
        sc = TParameterSummaryComponent(pos) and result = "Parameter[" + pos.toSpec() + "]"
```

Reply:
Good idea.